### PR TITLE
Document that attribute methods better make their results immutable

### PR DESCRIPTION
**Pull request by ThomasBreuer**

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]

File: doc/ref/types.xml [hunk 1]
```diff
@@ -435,10 +435,18 @@ computed, and claim that it knows these values.  Note that <Q>store</Q>
...
 (see&nbsp;<Ref Sect="Mutability and Copyability"/>),
 except if the attribute had been specially constructed as
-<Q>mutable attribute</Q>.
+<Q>mutable attribute</Q>, see <Ref Func="NewAttribute"/>.
+<P/>
+Note that functions installed as methods for attributes may return mutable
+objects,
+the attribute returns the value of <Ref Func="Immutable"/> for the result.
+Thus it is recommended that these methods make their return values immutable
+by calling <Ref Func="MakeImmutable"/> if this is possible,
```
Comment (fingolfin) [inline doc/ref/types.xml:447]: Maybe slightly more defensive (there are other ways to get an immutable object):
```suggestion
Thus it is recommended that these methods ensure their return values are immutable
e.g. by calling <Ref Func="MakeImmutable"/> if this is possible,
```

Comment (ThomasBreuer) [reply doc/ref/types.xml:447]: O.k.

File: doc/ref/types.xml [hunk 2]
```diff
@@ -435,10 +435,18 @@ computed, and claim that it knows these values.  Note that <Q>store</Q>
 and <Q>know</Q> have to be understood in the sense that it is very cheap
 to get such a value when the attribute is called again.
 <P/>
-The stored value of an attribute is in general immutable
+The value returned by the call of an attribute is in general immutable
```
Comment (james-d-mitchell) [inline doc/ref/types.xml:438]:
```suggestion
The value returned by an attribute is in general immutable
```

Comment (ThomasBreuer) [reply doc/ref/types.xml:438]: O.k., thanks.

Comment (james-d-mitchell) [reply doc/ref/types.xml]: Aha, ok, then better with just "returned by an attribute" then.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**